tests/main/debug-execution/task.yaml

```diff
@@ -0,0 +1,163 @@
+
+            echo "Checking without reeexec scenario"
+            MATCH 'distro-supports-reexec: true' < snap-no-reexec.out
+            MATCH 'is-reexec-enabled: false' < snap-no-reexec.out
+            MATCH 'is-reexec-explicitly-enabled: false' < snap-no-reexec.out
+            MATCH 'is-reexecd: false' < snap-no-reexec.out
+            MATCH 'self-exe: /usr/bin/snap' < snap-no-reexec.out
+
+            echo "Checking AppArmor"
+            # TODO: once snapd snap lands, this will use internal parser
+            # MATCH 'apparmor-parser: /snap/snapd/.*/usr/lib/snapd/apparmor_parser' < snap-apparmor-default.out
+            # MATCH 'internal: true' < snap-apparmor-default.out
+            if os.query is-xenial || os.query is-bionic; then
+                # Ubuntu < 20.04 does not have usr-merge
+                MATCH 'apparmor-parser: /sbin/apparmor_parser' < snap-apparmor-default.out
+                MATCH 'internal: false' < snap-apparmor-default.out
+                MATCH 'apparmor-parser: /sbin/apparmor_parser' < snap-apparmor-no-reexec.out
+                MATCH 'internal: false' < snap-apparmor-no-reexec.out
+            else
+                MATCH 'apparmor-parser: /usr/sbin/apparmor_parser' < snap-apparmor-default.out
+                MATCH 'internal: false' < snap-apparmor-default.out
+                MATCH 'apparmor-parser: /usr/sbin/apparmor_parser' < snap-apparmor-no-reexec.out
+                MATCH 'internal: false' < snap-apparmor-no-reexec.out
+            fi
+
+            # TODO: once snapd snap lands, this will use snap-update-ns from
+            # snapd snap
+            MATCH 'snap-update-ns: /snap/core/.*/usr/lib/snapd/snap-update-ns' < snap-uns-default.out
+            ;;
+        *)
+            echo "Checking default scenario"
+            MATCH 'distro-supports-reexec: false' < snap-default.out
+            MATCH 'is-reexec-enabled: true' < snap-default.out
+            MATCH 'is-reexec-explicitly-enabled: false' < snap-default.out
+            MATCH 'is-reexecd: false' < snap-default.out
+            MATCH 'self-exe: /usr/bin/snap' < snap-default.out
+
+            echo "Checking with reexec scenario"
+            MATCH 'distro-supports-reexec: false' < snap-yes-reexec.out
+            MATCH 'is-reexec-enabled: true' < snap-yes-reexec.out
+            MATCH 'is-reexec-explicitly-enabled: true' < snap-yes-reexec.out
+            # actual outcome depends on whether there is a /snap ->
+            # /var/lib/snapd/snap symlink on systems where snaps are not mounted
+            # under /snap
+
+            # TODO: once snapd snap lands the output wlll be different:
+            # MATCH 'is-reexecd: false' < snap-yes-reexec.out
+            # MATCH 'self-exe: /snap/snapd/.*/usr/bin/snap' < snap-yes-reexec.out
+            case "$SPREAD_SYSTEM" in
+                arch-linux-*|fedora-*|centos-*)
+                    # no /snap -> /var/lib/snapd/snap symlink by default
```

Comment:
I wonder if it would be easy and interesting to also test when the symlink is created.

Also should check right here that `/snap` does not exist? What if for some reason, the symlink exists (because of another test for example)? Then we would errors in the test. But the error would be that the assumption the symlink does not exist is wrong.

Reply:
I think that's fine. If the test fails, we'll see a list of tests which executed right before it and we'll know which one did not clean up.